### ISSUE TITLE
puppeteer executablePath doesn't seem to be honored (on windows)

### ISSUE THREAD

**Issue by jonmadison** [puppeteer/puppeteer#3013]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version:
    "puppeteer": "^1.4.0",
    (also tried 1.6.x) 

* Platform / OS version: WIndows 10
* URLs (if applicable):
* Node.js version: 10.5.0

**What steps will reproduce the problem?**

1. in code, set executablePath to an existing chromium browser as an option
(`browserpath` set earlier as "C:\\Users\\jonmadi\\Documents\\chrome-win32\\chrome.exe")
```
  let browserOptions = {
                          executablePath: browserpath,
                          headless: headless,
                          slowMo: speed
                       }

  log.debug(`browserOptions: ${JSON.stringify(browserOptions)}`)
  log.debug(`user: ${JSON.stringify(user)}`)
  //step 1 launch browser
  process.env['PUPPETEER_SKIP_CHROMIUM_DOWNLOAD'] = true
  const browser = await puppeteer.launch(browserOptions);
```

I've also tried setting PUPPETEER_SKIP_CHROMIUM_DOWNLOAD = true on the command line
and for giggles, "false".

2. run code, puppeteer launch executes

**What is the expected result?**
puppeteer would launch and begin to take screenshots

**What happens instead?**
get error that chromium is not installed.

log output:
``` browserOptions: {"executablePath":["C:\\Users\\jonmadi\\Documents\\chrome-win32\\chrome.exe"],"headless":false,"slowMo":0}
{ AssertionError [ERR_ASSERTION]: Chromium revision is not downloaded. Run "npm install" or "yarn install"
    at Console.assert (console.js:191:23)
    at Function.launch (E:\Automation\src\NodePageScreenshotTool\node_modules\puppeteer\lib\Launcher.js:100:15)
    at <anonymous>
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '==' }
```

interestingly enough, it runs on mac, but the excutablePath() function *after* launching does not reflect the custom set browser, but the one inside node_modules.

**Comment by aslushnikov** [comment]
@jonmadison how do you define the `browserpath` variable? It indeed won't work if you create it as a String object rather than string literal, e.g. `new String('/usr/var/...')`

**Comment by jonmadison** [comment]
I cannot tell you how many times this sort of thing has bitten me, in particular with the puppeteer API so i should know better by now. :( was able to fix with your response.

BTW I was using a template string.

**Comment by aslushnikov** [comment]
Thanks @jonmadison,

This is now fixed on tip-of-tree with #2950.
